packages/web/src/components/comments/TimestampLink.tsx

```diff
@@ -1,24 +1,60 @@
 import { useCurrentCommentSection } from '@audius/common/context'
-import { seek } from '@audius/common/src/store/player/slice'
-import { TextLink, formatTrackTimestamp } from '@audius/harmony'
-import { useDispatch } from 'react-redux'
+import { useGatedContentAccess } from '@audius/common/hooks'
+import { ModalSource } from '@audius/common/models'
+import {
+  playerActions,
+  PurchaseableContentType,
+  trackPageLineupActions,
+  trackPageSelectors,
+  usePremiumContentPurchaseModal
+} from '@audius/common/store'
+import { formatTrackTimestamp, TextLink, TextLinkProps } from '@audius/harmony'
+import { useDispatch, useSelector } from 'react-redux'
+
+const { tracksActions } = trackPageLineupActions
+const { getLineup } = trackPageSelectors
+const { seek } = playerActions
+
+type TimestampLinkProps = {
+  timestampSeconds: number
+} & TextLinkProps
+
+export const TimestampLink = (props: TimestampLinkProps) => {
+  const { timestampSeconds, ...other } = props
 
-export const TimestampLink = ({
-  trackTimestampS
-}: {
-  trackTimestampS: number
-}) => {
   const dispatch = useDispatch()
-  const { playTrack } = useCurrentCommentSection()
-  const handleClick = () => {
-    // Starts playing the current page's song
-    playTrack()
-    // Seeks to the timestamp
-    dispatch(seek({ seconds: trackTimestampS }))
-  }
+  const { track } = useCurrentCommentSection()
+  const lineup = useSelector(getLineup)
+  const { track_id: trackId } = track
+
+  const { hasStreamAccess } = useGatedContentAccess(track)
+
+  const uid = lineup?.entries?.[0]?.uid
+  const { onOpen: openPremiumContentPurchaseModal } =
+    usePremiumContentPurchaseModal()
+
   return (
-    <TextLink size='xs' variant='active' onClick={handleClick}>
-      {formatTrackTimestamp(trackTimestampS)}
+    <TextLink
+      onClick={() => {
+        if (!hasStreamAccess) {
+          openPremiumContentPurchaseModal(
+            { contentId: trackId, contentType: PurchaseableContentType.TRACK },
+            {
+              source: ModalSource.Comment
+            }
+          )
+        } else {
+          dispatch(tracksActions.play(uid))
+          setTimeout(() => {
```

Comment:
side note; we should probably clean this up from a wholistic level at some point. So silly that we cant queue a track & seek to a time as part of a single action and just expect it to work lol

Reply:
yeah I agree, crazy that we haven't needed this before

Reply:
The play action should just accept a timestamp